### ISSUE TITLE
External middleware does not compile: stop vendoring: miekg/dns and context

### ISSUE THREAD

**Issue by miekg** [coredns/coredns#792]
While trying to get the external middleware *alias* going, I needed to remove two vendored directories
from both repos to make it work:

* `rm -rf vendor/github.com/miekg/dns`
* `rm -rf vendor/golang.org/x/net/context`

Would should probably stop vendoring these.

**Comment by miekg** [comment]
Update all deps leads at least to this compile error:
~~~
# github.com/coredns/coredns/vendor/github.com/openzipkin/zipkin-go-opentracing/thrift/gen-go/scribe
vendor/github.com/openzipkin/zipkin-go-opentracing/thrift/gen-go/scribe/scribe.go:333: cannot use scribeProcessorLog literal (type *scribeProcessorLog) as type thrift.TProcessorFunction in assignment:
	*scribeProcessorLog does not implement thrift.TProcessorFunction (wrong type for Process method)
		have Process(context.Context, int32, thrift.TProtocol, thrift.TProtocol) (bool, thrift.TException)
		want Process(int32, thrift.TProtocol, thrift.TProtocol) (bool, thrift.TException)
vendor/github.com/openzipkin/zipkin-go-opentracing/thrift/gen-go/scribe/scribe.go:341: too many arguments in call to processor.Process
	have (context.Context, int32, thrift.TProtocol, thrift.TProtocol)
	want (int32, thrift.TProtocol, thrift.TProtocol)
86.26s user 4.66s system 37.21s elapsed 244%CPU ()
~~~

**Comment by miekg** [comment]
Ok, double testing with @serverwentdown's repo only makes the context.Context error show up. Can't get 'go dep' to *not* vender that dependency.

**Comment by miekg** [comment]
Grr: build barfs on coverage something
~~~
echo "" > coverage.txt
for d in `go list ./... | grep -v vendor`; do \
		go test -v  -tags 'etcd k8s' -race -coverprofile=cover.out -covermode=atomic -bench=. $d || exit 1; \
		if [ -f cover.out ]; then \
			cat cover.out >> coverage.txt; \
			rm cover.out; \
		fi; \
	done
vendor/github.com/coreos/etcd/client/discover.go:18:2: cannot find package "github.com/coreos/etcd/pkg/srv" in any of:
	/home/travis/gopath/src/github.com/coredns/coredns/vendor/github.com/coreos/etcd/pkg/srv (vendor tree)
	/home/travis/.gimme/versions/go1.8.linux.amd64/src/github.com/coreos/etcd/pkg/srv (from $GOROOT)
	/home/travis/gopath/src/github.com/coreos/etcd/pkg/srv (from $GOPATH)
vendor/github.com/coreos/etcd/client/client.go:31:2: cannot find package "github.com/coreos/etcd/version" in any of:
	/home/travis/gopath/src/github.com/coredns/coredns/vendor/github.com/coreos/etcd/version (vendor tree)
	/home/travis/.gimme/versions/go1.8.linux.amd64/src/github.com/coreos/etcd/version (from $GOROOT)
	/home/travis/gopath/src/github.com/coreos/etcd/version (from $GOPATH)
make: *** [coverage] Error 1
~~~
Can't now see if the build is broken or just this bit.

**Comment by miekg** [comment]
If we want to stop vender context, we'll also need to stop vendor these:
```
[[projects]]
  branch = "master"
  name = "golang.org/x/net"
  packages = ["context/ctxhttp","http2","http2/hpack","idna","internal/timeseries","lex/httplex","trace"]
  revision = "ab5485076ff3407ad2d02db054635913f017b0ed"
```
packages (I think).

**Comment by miekg** [comment]
Fixed with #794
